.github/workflows/spread-results-reporter.yaml

```diff
@@ -0,0 +1,123 @@
+            });
+            let matchingArtifacts = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name.startsWith(`spread-json-${context.payload.workflow_run.id}-${context.payload.workflow_run.run_attempt}`);
+            });
+            for (let artifact of matchingArtifacts) {
+              let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: artifact.id,
+                archive_format: 'zip',
+              });
+              let fs = require('fs');
+              fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/${artifact.name}.zip`, Buffer.from(download.data));
+              console.log(`Downloaded artifact: ${artifact.name}.zip`);
+            }
+ 
+      - name: Unzip spread json files
+        run: |
+          find . -name "spread-json-${{ github.event.workflow_run.id }}*.zip" | while read filename; do 
+            dir="${filename%.zip}"
+            mkdir $dir
+            unzip $filename -d $dir
+          done
+
+      - name: Echo collected output
+        run: |
+          # generate report
+          (
+          date
+          if ! ls spread-json-${{ github.event.workflow_run.id }}-*/*.json &> /dev/null; then
+            echo "## No spread failures"
+          else
+            jq -s 'add' spread-json-${{ github.event.workflow_run.id }}*/*.json > consolidated-report.json
+
+            echo "## Failures:"
+            if [[ $(jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "preparing" )' consolidated-report.json) ]]; then
+              echo "### Prepare:"
+              jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "preparing" ) .task' consolidated-report.json |\
+                  awk ' { print "- " $0 }'
+            fi
+            if [[ $(jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "executing" )' consolidated-report.json) ]]; then
+              echo "### Executing:"
+              jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "executing" ) .task' consolidated-report.json |\
+                  awk ' { print "- " $0 }'
+            fi
+            if [[ $(jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "restoring" )' consolidated-report.json) ]]; then
+              echo "### Restoring:"
+              jq -r '.[] | select( .info_type == "Error" ) | select( .verb == "restoring" ) .task' consolidated-report.json |\
+                  awk ' { print "- " $0 }'
+            fi
+          fi
+          ) > report
+          # display the report
+          grep -n '' report
+
+      - name: Comment report to PR
+        run: |
+          if ! gh pr comment "$(cat pr_number)" --body "$(cat report)" --edit-last; then
+            gh pr comment "$(cat pr_number)" --body "$(cat report)"
+          fi
+
+      - name: Delete artifacts
```

Comment:
Would those artifacts expire at some point? I wonder if we could just leave the artifact be?